src/embind/embind.js

```diff
@@ -660,25 +683,9 @@ var LibraryEmbind = {
   },
 
   _embind_register_emval__deps: [
-    '_emval_decref', '$Emval',
-    '$readLatin1String', '$registerType', '$simpleReadValueFromPointer'],
+    '$registerType',  '$EmValType'],
   _embind_register_emval: (rawType, name) => {
-    name = readLatin1String(name);
-    registerType(rawType, {
-      name,
-      'fromWireType': (handle) => {
-        var rv = Emval.toValue(handle);
-        __emval_decref(handle);
-        return rv;
-      },
-      'toWireType': (destructors, value) => Emval.toHandle(value),
-      'argPackAdvance': GenericWireTypeSize,
-      'readValueFromPointer': simpleReadValueFromPointer,
-      destructorFunction: null, // This type does not need a destructor
-
-      // TODO: do we need a deleteObject here?  write a test where
-      // emval is passed into JS via an interface
-    });
+    registerType(rawType, EmValType);
```

Comment:
I don't see `name` used anywhere here?  Is it really not needed?  If so I guess we can remove the argument completely?